schema-util/json/src/main/java/io/apicurio/registry/rules/compatibility/jsonschema/JsonUtil.java

```diff
@@ -39,57 +47,102 @@ public class JsonUtil {
         MAPPER.registerModule(new Jdk8Module());
         MAPPER.registerModule(new JavaTimeModule());
         MAPPER.registerModule(new JsonOrgModule());
-        MAPPER.enable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS);
+        MAPPER.enable(USE_BIG_DECIMAL_FOR_FLOATS);
         MAPPER.disable(FAIL_ON_UNKNOWN_PROPERTIES);
         MAPPER.setNodeFactory(JsonNodeFactory.withExactBigDecimals(true));
     }
 
-    public static Schema readSchema(String content) throws JsonProcessingException {
-        return readSchema(content, Collections.emptyMap(), true);
+    public static void readSchema(String content, Map<String, TypedContent> resolvedReferences)
+            throws JsonProcessingException {
+        readSchema(content, resolvedReferences, true);
     }
 
-    public static Schema readSchema(String content, Map<String, TypedContent> resolvedReferences)
-            throws JsonProcessingException {
-        return readSchema(content, resolvedReferences, true);
+    public static SomeJsonSchema readSchema(String content, Map<String, TypedContent> resolvedReferences,
+                                            boolean validateDangling) throws JsonProcessingException {
+
+        var jsonNode = MAPPER.readTree(content);
+
+        var specVersion = detect(jsonNode);
+
+        if (specVersion == UNKNOWN) {
+            // TODO: Make this configurable? Throw an exception?
```

Comment:
Leftover or intended?

Reply:
Intended as a note for possible future improvement.